src/sentry/data_export/tasks.py

```diff
@@ -21,157 +34,206 @@
 logger = logging.getLogger(__name__)
 
 
-@instrumented_task(name="sentry.data_export.tasks.assemble_download", queue="data_export")
+@instrumented_task(
+    name="sentry.data_export.tasks.assemble_download",
+    queue="data_export",
+    default_retry_delay=30,
+    max_retries=3,
+)
 def assemble_download(
     data_export_id,
     export_limit=1000000,
     batch_size=SNUBA_MAX_RESULTS,
+    offset=0,
+    bytes_written=0,
     environment_id=None,
     **kwargs
 ):
-    # Get the ExportedData object
     try:
-        logger.info("dataexport.start", extra={"data_export_id": data_export_id})
-        metrics.incr("dataexport.start", tags={"success": True}, sample_rate=1.0)
+        if offset == 0:
+            logger.info("dataexport.start", extra={"data_export_id": data_export_id})
+            metrics.incr("dataexport.start", tags={"success": True}, sample_rate=1.0)
+        logger.info("dataexport.run", extra={"data_export_id": data_export_id, "offset": offset})
         data_export = ExportedData.objects.get(id=data_export_id)
     except ExportedData.DoesNotExist as error:
-        metrics.incr("dataexport.start", tags={"success": False}, sample_rate=1.0)
+        if offset == 0:
+            metrics.incr("dataexport.start", tags={"success": False}, sample_rate=1.0)
         capture_exception(error)
```

Comment:
> Only tangentially related to this pr, but wanted to check whether you have plans to use the kafka `offset` value to pin the queries at some point. This will help with pagination, and should stop rows from jumping to the top and changing the order.

Is this offset not the same offset in `discover.query`?